### PR TITLE
refine go ps directory

### PR DESCRIPTION
**Pull request by QiJune**
Add a cmd directory, named `start_ps`.  As a result, the Go PS binary name is also changed to `start_ps`.

### CODE REVIEW

File: elasticdl/go/cmd/start_ps/main.go [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
 
 import (
-	"elasticdl.org/elasticdl/common"
-	"elasticdl.org/elasticdl/ps"
 	"flag"
 	"fmt"
 	"log"
 	"os"
 	"time"
+
```
Comment (workingloong) [inline elasticdl/go/cmd/start_ps/main.go:22]: Is the empty line needed?

File: elasticdl/go/pkg/common/tensor.go [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+
 package common
 
 import (
-	"elasticdl.org/elasticdl/proto"
 	"fmt"
+	"reflect"
+	"unsafe"
+
+	"elasticdl.org/elasticdl/pkg/proto"
```
Comment (workingloong) [inline elasticdl/go/pkg/common/tensor.go:21]: "elasticdl.org/elasticdl/go/pkg/proto"?

File: elasticdl/go/cmd/start_ps/main.go (hunk 1 above)
Comment (QiJune) [reply elasticdl/go/cmd/start_ps/main.go:22]: It's formated by visual code automatically.

File: elasticdl/go/pkg/common/tensor.go (hunk 2 above)
Comment (QiJune) [reply elasticdl/go/pkg/common/tensor.go:21]: Here is the reference for go project structure, https://github.com/golang-standards/project-layout.  A project has pkg and cmd.

We build the go project by a Dockerfile of elasticdl dev.
```
cp -r /elasticdl/elasticdl/go/ ${ELASTICDLPATH}
cd $ELASTICDLPATH
go mod init elasticdl.org/elasticdl
go mod tidy
go install ./...
```

The `go` directory is copied and renamed to a new directory, and we use go module to init a new module in this new directory, named `elasticdl.org/elasticdl`

So the import path is `elasticdl.org/elasticdl/pkg/proto`

**Comment by workingloong** [comment]
LGTM

**Review by workingloong — APPROVED** [review]
LGTM

**Comment by wangkuiyi** [comment]
> Add a cmd directory, named `start_ps`. As a result, the Go PS binary name is also changed to `start_ps`.

How about the name `elasticdl_ps`, or simply `ps`? @QiJune

**Pull request merged**